### ISSUE TITLE
Bug:  StaticPublisher::republish() doesn't strip slashes from $legalPages

### ISSUE THREAD

**Issue by Martimiz** [silverstripe-labs/silverstripe-staticpublisher#61]
If pagesAffectedByChanges() end allPagesToCache() both return urls with a trailing slash, then StaticPublisher::republish() wil strip that slash from its $urls, but not from its $legalPages. So intersecting $urls and $legalPages will allways return an empty array.

This afflicts master and v1.1.0

**Solution:** strip the slashes from $legalPages as well. 

**Temporary workaround**: make sure your pagesAffectedByChanges() end allPagesToCache() return $urls without trailing slash :)

**Comment by Martimiz** [comment]
Yeah, I referenced apull request by mistake, and changed that, but it still hangs around here. Cache issue??? If it's still here when you visit - please ignore! :)

**Comment by dhensby** [comment]
@Martimiz would you mind submitting a PR?
